autotest/gcore/tiff_write.py

```diff
@@ -10256,5 +10257,72 @@ def test_tiff_write_createcopy_alpha_not_in_last_band(options):
     gdal.GetDriverByName("GTiff").Delete(tmpfilename)
 
 
+###############################################################################
+# Test JXL compression
+def test_tiff_write_jpegxl_band_combinations():
+
+    tmpfilename = "/vsimem/test_tiff_write_jpegxl_band_combinations.tif"
+    md = gdal.GetDriverByName("GTiff").GetMetadata()
+    if md["DMD_CREATIONOPTIONLIST"].find("JXL") == -1:
+        pytest.skip()
+
+    
+    src_ds = gdal.GetDriverByName("MEM").Create("", 64, 64, 6)
+    for b in range(6):
+        bnd = src_ds.GetRasterBand(b+1)
+        data = numpy.random.bytes(64*64)
+        bnd.WriteRaster(0,0,64,64,data)
+        #bnd.Fill(b+1)
```

Comment:
@rouault if I use `bnd.Fill` instead of numpy.random + writeraster, the checksum of the band will always be 0. I would prefer to not have to import numpy here and rely on the checksum varying depending on the fill value, is that possible?

Reply:
weird that checksum is 0. maybe try to add bnd.FlushCache() after bnd.Fill() ? But that shouldn't be necessary

Reply:
Works for me without FlushCache():
```
$ python -c "from osgeo import gdal; ds = gdal.GetDriverByName('MEM').Create('', 64, 64); ds.GetRasterBand(1).Fill(1); print(ds.GetRasterBand(1).Checksum())"
4096
```